### PR TITLE
Update config file logic, tests

### PR DESCRIPTION
**Pull request by mjn0898**
Allowing config file object to be passed into __init__

### CODE REVIEW

File: examples/example_mineral.py [hunk 1]
```diff
@@ -70,9 +70,14 @@ def run_mineral(input_filename=constants.INPUT_FILENAME,
 
     # path to save classification scores image
     scores_filename = constants.INPUT_NAME + "_scores.hdr"
-
+    
+    # path to config.ini file w/ configuration parameters
+    config_filename = os.path.join(os.path.dirname(__file__),  "..", "pycoal", "config.ini")
```
Comment (lewismc) [inline examples/example_mineral.py]: This logic shoud not exist in the example... instead it should be pushed into the implementing Class invocation e.g.  `MineralClassification`

**Review by lewismc — CHANGES_REQUESTED** [review]
Please address comments
Also, please provide unit tests for the following
* incorrect paths
* garbled, garbage config values
Thanks

**Comment by mjn0898** [comment]
The tests are in progress, I just didn't have time to do them today. 
I am a bit confused on wanting that logic in the class invocation because you wanted the config file to be passed in as a parameter to the __init__ function, which is why I now pass in a file object which I have to create elsewhere.

**Comment by lewismc** [comment]
> The tests are in progress, I just didn't have time to do them today.

No problem. That's fine. I understand.

> I am a bit confused on wanting that logic in the class invocation because you wanted the config file to be passed in as a parameter to the init function, which is why I now pass in a file object which I have to create elsewhere.

Here's what I think would make a sensible API
1) augment the pycoal-mineral CLI [options](https://github.com/capstone-coal/pycoal/blob/master/pycoal/cli/mineral.py#L43-L54) to include an option for a configuration file,
2) augment the [MineralClassification __init__ function](https://github.com/capstone-coal/pycoal/blob/master/pycoal/mineral.py#L315) such that, again, it includes an option for a configuration file. In this case the `os.path.join(os.path.dirname(__file__)...` logic should reside within `MineralClassification __init__` because it may not be called from the example, it may be called from somewhere else e.g. another Python script or pycoal-mineral CLI.

Does that make sense?

**Comment by lewismc** [comment]
Also, please always label your issues correctly and assign them to the current project. Thanks @mjn0898  good work.

File: pycoal/cli/mineral.py [hunk 2]
```diff
@@ -52,6 +52,8 @@ def main():
     parser.add_argument("-c", "--classified_filename",
                         dest="classified_filename",
                         help="Classified File Name")
+    parser.add_argument("-cf", "--config_filename", dest="config_filename",
+                        help="Configuration File Name")
```
Comment (lewismc) [inline pycoal/cli/mineral.py]: The help message could be improved here detailing what the options are e.g. relative, absolute and what the  default is.

File: pycoal/mineral.py [hunk 3]
```diff
@@ -312,7 +312,7 @@ def avngDNN(image_file_name, classified_file_name, model_file_name,
 
 class MineralClassification:
 
-    def __init__(self, algorithm=SAM_pytorch, **kwargs):
+    def __init__(self, algorithm=SAM_pytorch, config_file=None, **kwargs):
```
Comment (lewismc) [inline pycoal/mineral.py]: Is it `None` or is it `config.uni`?

**Review by lewismc — CHANGES_REQUESTED** [review]
Thanks for the PR, please see my comments.

**Comment by lewismc** [comment]
Also, @mjn0898 please address the DeepSource issues. Thanks

**Comment by mjn0898** [comment]
I fixed the deepsource issues. As for the build errors due to some new tests failing, most of the errors are due to the new algorithm functions not being up to date on this branch.

**Comment by lewismc** [comment]
@mjn0898 this is looking much better.

> As for the build errors due to some new tests failing, most of the errors are due to the new algorithm functions not being up to date on this branch.

Well then please just use the following annotation for the time being
```
@unittest.skip("reason for skipping")
```
See the follosing SO post - https://stackoverflow.com/questions/2066508/disable-individual-python-unit-tests-temporarily
Thanks

**Comment by lewismc** [comment]
Thank you @mjn0898 good work on this branch. Some notes for your consideration
* please always create an issue before you create the pull request. This should describe the scope of the work. Your commit messages should also reflect this scope.
* on commit messages, please see the following - https://chris.beams.io/posts/git-commit/

Thank you, good work

**Comment by mjn0898** [comment]
Thanks, I will keep that in mind

**Pull request merged**